### PR TITLE
fix documentation links broken by OMERO decoupling

### PR DESCRIPTION
**Pull request by mtbc**
Trying to get [OMERO-DEV-merge-docs](https://ci.openmicroscopy.org/job/OMERO-DEV-merge-docs/) passing. See https://github.com/openmicroscopy/openmicroscopy/pull/5934 and https://trello.com/c/ZVrR6L3F/91-build-system-changes.

### CODE REVIEW

**Comment by mtbc** [comment]
Given how much this PR touches I suggest that we merge it promptly and other OMERO 5.5.0 documentation fixes can follow separately.

File: omero/developers/Matlab.rst [hunk 1]
```diff
@@ -267,7 +267,7 @@ argument::
   memory-consuming depending on the amount of data.
 
 To return the orphaned datasets i.e. datasets not in a project, as well as the projects, you can query the second output argument of
-:source:`getProjects <components/tools/OmeroM/src/io/getProjects.m>`::
+:matlab_source:`getProjects <components/tools/OmeroM/src/io/getProjects.m>`::
```
Comment (joshmoore) [inline omero/developers/Matlab.rst:270]: Should this not drop `components/tools/OmeroM/`?

File: omero/developers/Matlab.rst [hunk 2]
```diff
@@ -741,15 +741,15 @@ image ``image_id``::
...
+:matlab_source:`updateFileAnnotation <components/tools/OmeroM/src/annotations/updateFileAnnotation.m>` function.
 If the file annotation has been retrieved from the server as
 ``fileAnnotation``, then the content of its ``OriginalFile`` can be replaced
 by the content of ``local_file_path`` using::
 
     updateFileAnnotation(session, fileAnnotation, local_file_path);
 
 .. seealso::
-  :source:`WriteData.m <examples/Training/matlab/WriteData.m>`
+  :matlab_source:`WriteData.m <examples/Training/matlab/WriteData.m>`
```
Comment (joshmoore) [inline omero/developers/Matlab.rst:752]: I didn't extract these (though maybe I should have)

**Review by joshmoore — COMMENTED** [review]
Few minor points around matlab. Otherwise, :+1: for the strategy and happy to get this in sooner rather than later and iterate.

File: omero/developers/Matlab.rst (hunk 1 above)
Comment (mtbc) [reply omero/developers/Matlab.rst:270]: I too had expected so but no: see https://github.com/ome/omero-matlab/tree/master/components/tools/OmeroM

File: omero/developers/Matlab.rst (hunk 2 above)
Comment (mtbc) [reply omero/developers/Matlab.rst:752]: Certainly could see the examples going to a different repo.

**Comment by mtbc** [comment]
@joshmoore: If you rearrange the MATLAB stuff happy to do a followup PR: that was by far one of the least painful aspects of this doc-fixing!

File: omero/developers/Matlab.rst (hunk 1 above)
Comment (joshmoore) [reply omero/developers/Matlab.rst:270]: Ah, see https://github.com/ome/omero-matlab/pull/9 cc: @jburel @sbesson How do we want to order all of this?

Comment (sbesson) [reply omero/developers/Matlab.rst:270]: From my side, I certainly approve these changes - see https://github.com/ome/omero-matlab/pull/9#pullrequestreview-160657146 but I understand this has not been tested yet and `omero-matlab` will not be the focus of the immediate next 5.5.0 milestones.

I think we are down to 2 options:
- merging this PR as it is and handling the renaming and the documentation changes as part of the dedicated milestone
- merging https://github.com/ome/omero-matlab/pull/9 immediately and updating the links right now.

If the second option is preferable, I might be able to carry out some minimal testing and report issues directly on the `omero-matlab` repo that we can consider when we come back to it.

Comment (joshmoore) [reply omero/developers/Matlab.rst:270]: Thanks, @sbesson. It being Friday, let's err on the side of merging this PR as opposed merging 9, pushing commits, merging this PR, etc.

**Pull request merged**